src/libcore/fmt/mod.rs

```diff
@@ -529,6 +529,29 @@ pub trait Debug {
 #[stable(feature = "rust1", since = "1.0.0")]
 pub trait Display {
     /// Formats the value using the given formatter.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::fmt;
+    ///
+    /// enum Foo {
+    ///     Bar,
+    ///     Stack,
+    /// }
+    ///
+    /// impl fmt::Display for Foo {
+    ///     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+    ///         match *self {
+    ///             Foo::Bar => write!(f, "Foo::Bar"),
+    ///             Foo::Stack => write!(f, "Foo::Stack"),
```

Comment:
I always add the enum's name first in case you have multiple subcasing:

```rust
enum Foo {
    A,
}

enum X {
    A(Foo),
}

enum Y {
    A(X),
}
```